openrr-client/src/robot_client.rs

```diff
@@ -338,3 +337,37 @@ where
         self.speaker.speak(message)
     }
 }
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct OpenrrClientsConfig {
+    pub urdf_path: String,
+    urdf_full_path: Option<PathBuf>,
+    pub self_collision_check_pairs: Vec<String>,
+
+    pub collision_check_clients_configs: Vec<CollisionCheckClientConfig>,
+    pub ik_clients_configs: Vec<IkClientConfig>,
+
+    pub joints_poses: Vec<JointsPose>,
+}
+
+impl OpenrrClientsConfig {
+    pub fn resolve_path<P: AsRef<Path>>(&mut self, path: P) -> Result<(), Error> {
+        self.urdf_full_path = Some(
+            path.as_ref()
+                .parent()
+                .ok_or_else(|| Error::NoParentDirectory(path.as_ref().to_owned()))?
+                .join(&self.urdf_path),
+        );
+        Ok(())
+    }
+    pub fn urdf_full_path(&self) -> &Option<PathBuf> {
+        &self.urdf_full_path
+    }
```

Comment:
Comment:

I thought that it is better to generate `udf_full_path` if there are no path yet.

```rust
pub fn urdf_full_path(&self) -> Result<&PathBuf> {
    if self.urdf_full_path.is_none() {
       self.resolve_path(&self.urdf_path)?;
    }
    &self.urdf_full_path
}
```